### PR TITLE
get exits more reliably and get deeper nested paths

### PR DESCRIPTION
**Pull request by RafaelOrtizRC**
This updates our probes with the following features:
* get deeper nested paths from process starts
* get inode information more reliably
* capture process exits more reliably

### CODE REVIEW

File: src/process-events.c [hunk 1]
```diff
@@ -906,6 +950,74 @@ int kretprobe__ret_sys_clone3(struct pt_regs *ctx)
...
+    read_value(ptr, CRC_INODE_I_RDEV, &i_rdev, sizeof(i_rdev));
+    read_value(ptr, CRC_INODE_I_INO, &i_ino, sizeof(i_ino));
+
+    u64 *id = bpf_map_lookup_elem(&process_ids, &pid_tgid);
+
+    if (!id)
+        return -1;
+
+    // prepare event
+    ptelemetry_event_t ev = &(telemetry_event_t){
```
Comment (bjackson41) [inline src/process-events.c:986]: this looks a little awkward to me. are you forcing a pointer for macro use?

File: src/process-events.c [hunk 2]
```diff
@@ -367,10 +397,24 @@ Pwd:;
     SET_OFFSET(CRC_DENTRY_D_PARENT);
     u32 parent = *(u32 *)offset; // offset of d_parent
 
-    SEND_PATH_N(9);
-    bpf_tail_call(ctx, &tail_call_table, SYS_EXECVE_4_8);
+    u64 _to_skip = (u64)bpf_map_lookup_elem(&read_path_skip, &to_skip);
+    if (_to_skip) {
+        __builtin_memcpy(&to_skip, (void*)_to_skip, sizeof(u64));
+        SKIP_PATH_N(100);
```
Comment (bjackson41) [inline src/process-events.c:403]: whats the reasoning behind these SEND_PATH magic values?

Comment (RafaelOrtizRC) [reply src/process-events.c]: This number (I bumped it up to `150`) is determined experimentally, keeping us under `4096` instructions. Sending takes up far more instructions than skipping, so this felt like a good balance.

File: src/process-events.c (hunk 1 above)
Comment (RafaelOrtizRC) [reply src/process-events.c:986]: Pretty much. To get wide kernel version support, the structure must live on the stack but also we need to provide the helpers with pointers. We don't actually use the regular variable, so it just wastes stack and/or instructions. I use this pattern in a few places.

File: src/process-events.c [hunk 3]
```diff
@@ -906,6 +950,74 @@ int kretprobe__ret_sys_clone3(struct pt_regs *ctx)
...
+    read_value(ts, CRC_TASK_STRUCT_MM, &ptr, sizeof(ptr));
+    read_value(ptr, CRC_MM_STRUCT_EXE_FILE, &ptr, sizeof(ptr));
+    read_value(ptr, CRC_FILE_F_INODE, &ptr, sizeof(ptr));
+    read_value(ptr, CRC_INODE_I_RDEV, &i_rdev, sizeof(i_rdev));
+    read_value(ptr, CRC_INODE_I_INO, &i_ino, sizeof(i_ino));
+
+    u64 *id = bpf_map_lookup_elem(&process_ids, &pid_tgid);
+
+    if (!id)
+        return -1;
```
Comment (bjackson41) [inline src/process-events.c:983]: IIRC, from the WPM events, we always `return 0`. After reading https://github.com/iovisor/bcc/issues/139, it looks as though returning non-zero has some meaning for kprobes. I wanted to confirm that we are aware of this and returning `-1` was intentional.

Comment (RafaelOrtizRC) [reply src/process-events.c:983]: You're right - we should `return 0` in all cases here and handle partial events on the userspace side.

Comment (RafaelOrtizRC) [reply src/process-events.c]: 1c112bb298a8ebf0a28a70535c7755529958190c

**Review by bjackson41 — APPROVED** [review]
Looks to be working. Curious how much the `return 0;` made a difference, if any.

**Pull request merged**